tests/test_cli.py

```diff
@@ -734,3 +734,27 @@ def test_config_file(self):
 
                 self.assertEqual((ctx.returncode, ctx.stdout, ctx.stderr),
                                  (0, '', ''))
+
+    def test_parent_config_file(self):
+        workspace = { 'a/b/c/d/e/f/g/a.yml': 'hello: world\n' }
+        conf = ('---\n'
+                'extends: relaxed\n')
+
+        for conf_file in ('.yamllint', '.yamllint.yml', '.yamllint.yaml'):
+            with self.subTest(conf_file):
+                with temp_workspace(workspace):
+                    with RunContext(self) as ctx:
+                        os.chdir('a/b/c/d/e/f')
```

Comment:
I was surprised that calling `os.chdir()` here doesn't need to be "undone" after each iteration, but apparently it's OK, it's automatically done by `temp_workspace()`: https://github.com/adrienverge/yamllint/blob/019c87d/tests/common.py#L76-L86


```diff
@@ -741,10 +741,13 @@ class CommandLineConfigTestCase(unittest.TestCase):
                 'extends: relaxed\n')
 
         for conf_file in ('.yamllint', '.yamllint.yml', '.yamllint.yaml'):
+            print('starting test in ' + os.getcwd())
             with self.subTest(conf_file):
                 with temp_workspace(workspace):
+                    print('running subtest in ' + os.getcwd())
                     with RunContext(self) as ctx:
                         os.chdir('a/b/c/d/e/f')
+                        print('running context in ' + os.getcwd())
                         cli.run(('-f', 'parsable', '.'))
 
                 self.assertEqual((ctx.returncode, ctx.stdout, ctx.stderr),
```

Reply:
Indeed, I checked that temp_workdir goes back to the previous working directory, so this should be good as is I hope.